src/rules/prefer-repository-shorthand.ts

```diff
@@ -0,0 +1,99 @@
+import type ESTree from "estree";
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import { createRule } from "../createRule.js";
+
+const githubUrlRegex =
+	/^(?:git\+)?(?:ssh:\/\/git@|http?s:\/\/)?(?:www\.)?github\.com\//;
+
+const isGitHubUrl = (url: string) => githubUrlRegex.test(url);
+
+const cleanGitHubUrl = (url: string) =>
+	url.replace(githubUrlRegex, "").replace(/\.git$/, "");
+
+export default createRule({
+	create(context) {
+		return {
+			JSONProperty: (node) => {
+				if (
+					node.key.type === "JSONLiteral" &&
+					node.key.value === "repository" &&
+					node.parent.parent.parent.type === "Program"
+				) {
+					if (node.value.type === "JSONObjectExpression") {
+						const { properties } = node.value;
+						const typeProperty = properties.find(
+							(property) =>
+								property.key.type === "JSONLiteral" &&
+								property.key.value === "type",
```

Comment:
[Testing] Missing test case, I think: what happens if `"repository"` is missing a `"type"`.